### ISSUE TITLE
Delay when opening Unite

### ISSUE THREAD

**Issue by toupeira** [Shougo/unite.vim#986]
I recently upgraded Unite from commit c3265691a917aa22ae18ed422f255d5f0e43d2ec to 3a25f2a1992e0e39d62a2034590580df540d8b80 and noticed a big delay when fuzzy finding using `file_rec/async` in a folder with several thousands of files. This used to be almost instantenous but now Vim freezes for several seconds before Unite appears.

You can find my [Unite configuration here](https://github.com/toupeira/dotfiles/blob/908653a4e52b5cdbdcf605b41784d5f3b24f448a/vim/vimrc#L1019-L1113), I tried stripping it down to the bare minimum but still experienced the delay.

I'll also attach a profile log taken during one Unite call, if you search for high self times you'll find several methods related to filtering and sorting.

**Comment by toupeira** [comment]
Profile is uploaded at https://gist.github.com/toupeira/e7c16c4d01d2cbdd8867 (>1MB)

**Comment by Shougo** [comment]
Please upload your Vim version.
I think this problem is fixed in Vim 7.4.615.

https://github.com/Shougo/unite.vim/issues/723

**Comment by toupeira** [comment]
This was on 7.4.712, I just upgraded to 7.4.826 and it's still the same.

I'm on Debian sid amd64.

**Comment by Shougo** [comment]
Confirmed.  Unfortunatelly, I cannot fix it.
You should disable "matcher_hidden_file" and "sorter_length" features in file_rec source.

```vim
call unite#custom#source('file_rec/async', 'sorters', [])
call unite#custom#source('file_rec/async', 'matchers', ['converter_relative_word', 'matcher_default'])
```

**Comment by toupeira** [comment]
Thanks, that improved things a bit.

But I went back to the old commit 3a25f2a and confirmed that the delay was gone again, so I looked at the diff for sources/rec.vim and noticed the change in `g:unite_source_rec_async_command` which now needs to be a list. I was still using a string so my `ag` command was ignored and `find` used instead (looks like there's a warning for this but it didn't show up for me). If I convert it to a list though I don't seem to get any results at all from the `file_rec/async` source (after refreshing with Ctrl+L). My setting is as follows:

    let g:unite_source_rec_async_command = [ 'ag', '-l', '-g', '""', '--nocolor' ]

Is there something wrong with this, or is there some other configuration change I missed?
